spec/trains/trains_spec.lua

```diff
@@ -163,11 +160,16 @@ describe('Trains module', function()
         end)
 
         it('only adds new trains to the registry', function()
+
+            -- Will let you deal with it but your train registry was not
+            -- getting cleared from the preceding function before calling the function so
```

Comment:
The entire train spec probably needs a rewrite, but that can wait until the changes for the new train events are in place.